### PR TITLE
Propose DEBUG to constraint level for developer debugging of models

### LINKED ISSUE #536: Add a DEBUG constraint level
# User Story:

As a Metschema developer, in order effectively and efficiently add testing constraints without the ability to filter out their passing and failing out for production, user-facing applications and facilitate developer debugging without editing the models pre-release, I would like the ability to use `level="DEBUG"` for all supported constraints.

## Goals:

- [x] Support debugging of models quickly
- [x] Implement a solution that does not require commenting out one or more debugging constraints frequently before publication of models "downstream"

More conversation around this proposal can be found [in the Gitter.im/Matrix community chat channel](https://matrix.to/#/!AeQqwsjGyZLiOqwtTs:gitter.im/$dbKAQeMjHSuT1othuGjFfHlvNUaVkHh6cwzs-gTasFU?via=gitter.im&via=matrix.org).

## Dependencies:

None

## Acceptance Criteria

- [x] All [website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes in this issue have been updated. Changes to the website can be made in the docs/content directory of your branch.
- [x] A Pull Request (PR) is submitted that fully addresses the goals of this User Story. This issue is referenced in the PR.
- [x] The CI-CD build process runs without any reported errors on the PR. This can be confirmed by reviewing that all checks have passed in the PR.

{The items above are general acceptance criteria for all User Stories. Please describe anything else that must be completed for this issue to be considered resolved.}

### PR DESCRIPTION
**Pull request by aj-stein-nist**
# Committer Notes

Closes #536.

### All Submissions:

- [x] Have you followed the guidelines in our [Contributing](https://github.com/usnistgov/metaschema/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/usnistgov/metaschema/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Do all automated CI/CD checks pass?

### Changes to Core Features:

- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you updated all website](https://pages.nist.gov/metaschema) and readme documentation affected by the changes you made? Changes to the website can be made in the website/content directory of your branch.
- ~Have you written new tests for your core changes, as applicable?~ N/A for spec change, we don't have applicable tests for the core module at this time
- ~Have you included examples of how to use your new feature(s)?~ No, I could consider adding them to a tutorial if this request comes up.

### CODE REVIEW

File: schema/metaschema/metaschema-module-metaschema.xml
Comment (david-waltermire) [inline schema/metaschema/metaschema-module-metaschema.xml:1030]: Move this to after INFORMATIONAL, since these are listed in order of severity?

File: schema/xml/metaschema.xsd
Comment (david-waltermire) [inline schema/xml/metaschema.xsd:868]: Move this to after INFORMATIONAL, since these are listed in order of severity?

**Comment by david-waltermire** [comment]
Should this PR update the specification?

**Comment by aj-stein-nist** [comment]
> Should this PR update the specification?

Do we definite the enumeration of all the values in the spec? One moment, let me check that.

File: schema/xml/metaschema.xsd
Comment (aj-stein-nist) [reply schema/xml/metaschema.xsd:868]: Relocated as requested in https://github.com/usnistgov/metaschema/commit/6ac24990042fded121843e8ee9e51a4b988e71bb.

File: schema/metaschema/metaschema-module-metaschema.xml
Comment (aj-stein-nist) [reply schema/metaschema/metaschema-module-metaschema.xml:1030]: Relocated as requested in https://github.com/usnistgov/metaschema/commit/6ac24990042fded121843e8ee9e51a4b988e71bb.

**Comment by aj-stein-nist** [comment]
> > Should this PR update the specification?
> 
> Do we definite the enumeration of all the values in the spec? One moment, let me check that. (I know I avoided that in the tutorial.)

OK this has been done in the spec in https://github.com/usnistgov/metaschema/pull/537/commits/1c663f233cbe06dc3f6bff8dd03b8ae7c44a5aa2. You were right, not hard to do (let alone find).

My only outstanding question, barring community review async or in a weekly community meeting: should I be updating [the marked up web version of the module syntax XML representation](https://pages.nist.gov/metaschema/specification/syntax/)? `<constraint/>` has been rightfully empty for a bit. Should I be editing to add here, or more reasonably in a specific issue, or hold off on that for a bit?

With the changes integrated, I am marking this ready for review and wider public comment.

File: schema/metaschema/metaschema-module-metaschema.xml
Comment (david-waltermire) [inline schema/metaschema/metaschema-module-metaschema.xml:1033]:
```suggestion
                <enum value="DEBUG">A violation of the constraint represents a fault in the content that may warrant review by a developer when performing model or tool development.</enum>                
```

File: schema/xml/metaschema.xsd
Comment (david-waltermire) [inline schema/xml/metaschema.xsd:883]:
```suggestion
          <xs:documentation>A violation of the constraint represents a fault in the content that may
            warrant review by a developer when performing model or tool development.</xs:documentation>
```

File: website/content/specification/syntax/constraints.md
Comment (david-waltermire) [inline website/content/specification/syntax/constraints.md:61]:
```suggestion
If defined, a `@level` MUST have a value of either: `CRITICAL`, `ERROR`, `WARNING`, `INFORMATIONAL`, or `DEBUG`.
```

**Review by david-waltermire — CHANGES_REQUESTED** [review]
Provided some comments to address.

**Review by david-waltermire — APPROVED** [review]

**Comment by david-waltermire** [comment]
I'll merge this on 2/21/2024 if there is no additional feedback.

**Pull request merged**
